pkg/middleware/auth.go

```diff
@@ -161,6 +161,12 @@ func SnapshotPublicModeOrSignedIn(cfg *setting.Cfg) macaron.Handler {
 	}
 }
 
+func ReqNotSignedIn(c *models.ReqContext) {
+	if c.IsSignedIn && !c.IsAnonymous {
```

Comment:
I believe `c.IsSignedIn` is false when `c.IsAnonymous` is true according to [code](https://github.com/grafana/grafana/blob/75ff02a9fb6c06664171cde75ddf87336a9f1886/pkg/services/contexthandler/contexthandler.go#L144-L146) so think the following should be enough:

```suggestion
	if c.IsSignedIn {
```